src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderSentenceResourceImplIntTest.kt

```diff
@@ -0,0 +1,711 @@
+      webTestClient.delete()
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_SYSTEM_USER")))
+        .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+    }
+
+    @Test
+    fun `should return 403 when user does not have required role`() {
+      webTestClient.put()
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_BANANAS")))
+        .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+        .bodyValue(
+          """
+              {
+                "passed": true,
+                "date": "2018-12-31"
+              }
+          """,
+        )
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `should return 403 when user has VIEW_PRISONER_DATA role`() {
+      webTestClient.put()
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_VIEW_PRISONER_DATA")))
+        .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+        .bodyValue(
+          """
+              {
+                "passed": true,
+                "date": "2018-12-31"
+              }
+          """,
+        )
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isForbidden
+    }
+  }
+
+  @Nested
+  @DisplayName("DELETE /api/offender-sentences/booking/{bookingId}/home-detention-curfews/latest/checks-passed")
+  inner class ClearHDCChecksPassedFlag {
+    @Test
+    fun `should return success when user has SYSTEM_USER role`() {
+      webTestClient.delete()
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_SYSTEM_USER")))
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isNoContent
+    }
+
+    @Test
+    fun `should return 403 when user does not have required role`() {
+      webTestClient.delete()
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_BANANAS")))
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `should return 403 when user has VIEW_PRISONER_DATA role`() {
+      webTestClient.delete()
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_VIEW_PRISONER_DATA")))
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isForbidden
+    }
+  }
+
+  @Nested
+  @DisplayName("PUT /api/offender-sentences/booking/{bookingId}/home-detention-curfews/latest/approval-status")
```

Comment:
I just noticed there are a few cucumber tests relating to the curfew endpoints - and these are the ones that I am changing roles for.   So will do as separate PR